pynicotine/gtkgui/frame.py

```diff
@@ -754,10 +754,26 @@ def set_up_actions(self):
         self.application.add_action(action)
         self.application.set_accels_for_action("app.settings", ["<Primary>comma", "<Primary>p"])
 
-        action = Gio.SimpleAction.new("quit", None)
+        action = Gio.SimpleAction.new("quit", None)  # Menu 'Quit' always Quits
         action.connect("activate", self.on_quit)
         self.application.add_action(action)
-        self.application.set_accels_for_action("app.quit", ["<Primary>q"])
+
+        # Window (system menu and events)
+
+        action = Gio.SimpleAction.new("close", None)  # 'When closing Nicotine+'
+        action.connect("activate", self.on_close_request)
+        self.application.add_action(action)
+        self.application.set_accels_for_action("app.close", ["<Primary>q"])
+
+        action = Gio.SimpleAction.new("exit_dialog", None)  # Show all exit options
```

Comment:
Wouldn't it be better to have a shortcut that simply makes Nicotine+ run in the background instead?

Reply:
I tried this, but if the tray icon is not visible for some reason then it can lead to yet another situation where the application window disappears unexpectedly.

Until we have a separate Minimize to Tray function (that is able to show/hide the tray icon) then I feel a prompt is required incase the user pressed the wrong key.

Reply:
Isn't <Shift><Primary>q a rather obscure shortcut already? I don't foresee a situation where a user accidentally enables it.

If there's a concern that the user accidentally hides the window instead of quitting the application, perhaps the shortcut could use another key that is rarely used.

Reply:
> Isn't <Shift><Primary>q a rather obscure shortcut?... I don't foresee a situation where a user accidentally enables it. 

It is right next to `<Shift><Primary>w` (show Wishlist), and it is near to the numeric keys on the row above so it is possible to press Q accidentally whilst Shift and Ctrl are also pressed, especially if sticky keys are enabled.

> perhaps the shortcut could use another key

The only logical key I can think of would be some key modified `F4` combo but that is likely to get into key conflict territory. I have also implemented `<Primary><Alt>Q` for force-quit, so if the remembered choice is set to 'Run in Background' then it is possible do any of the three actions available in the 'When closing Nicotine+' options at any time.

> perhaps we don't need the shortcut at all?... it's probably unlikely that the user wants anything else than the remembered choice.

Right now this is the case, but in the future if the exit dialog gets more buttons or a dropdown for other exit options, such as "[Finish running transfers then quit](https://github.com/nicotine-plus/nicotine-plus/issues/885)", or other suchlike, then this shortcut would serve a useful purpose to be able to override the remembered choice with some other action.

Reply:
> Right now this is the case

Let's remove it for now. Options such as "Finish running transfers then quit" will require further design discussions.